### PR TITLE
Documentation homepage, use of panels for visibility to users

### PR DESCRIPTION
**Pull request by tkknight**
### Description

Added some panels and use of badges on the documentation index, in order to make it more visible to users.

- The link to the legacy docs was in a note, now moved to lower on the page and uses a badge link instead.
- Added panels using the [sphinx-panels](https://sphinx-panels.readthedocs.io/en/latest/) extension.
- Made the index on the main body hidden, now it only appears in the sidebar.
- Added the dependency for `sphinx-panels`.
- Moved the `Iris API` link in the sidebar to the Reference section

You can view all rendered changes at https://iris-test-doc.readthedocs.io/en/latest/ for the lifetime of this PR.

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

File: docs/iris/src/index.rst [hunk 1]
```diff
@@ -44,11 +38,61 @@ Interoperability with packages from the wider scientific Python ecosystem comes
...
+        :classes: btn-outline-primary btn-block
+    ---
+    View the gallery that includes python code used to create it.
+    +++
+    .. link-button:: sphx_glr_generated_gallery
+        :type: ref
+        :text: Gallery
+        :classes: btn-outline-primary btn-block
+    ---
+    Find out what has recently been added to Iris, or soon will be.
```
Comment (rcomer) [inline docs/iris/src/index.rst:63]: Not sure about the "soon will be", as this won't make sense for the released docs.  Also this is currently pointing to a list of all the whatsnews.  Should it just be the whatsnew for the current version?  If intended to point to all versions, maybe "Find out what has changed in Iris" would be more all-encompassing.

**Comment by tkknight** [comment]
any thoughts @SciTools/iris-devs?

**Comment by pp-mo** [comment]
:+1: really like the improved clarity of this (and incorporating #3880).
Can't find anything to fault with this, but I will leave @bjlittle to complete review

File: docs/iris/src/index.rst [hunk 2]
```diff
@@ -44,11 +38,61 @@ Interoperability with packages from the wider scientific Python ecosystem comes
...
+    +++
+    .. link-button:: iris_whatsnew
+        :type: ref
+        :text: What's new
```
Comment (rcomer) [inline docs/iris/src/index.rst:67]: Capital N for consistency with the other buttons?

File: docs/iris/src/index.rst [hunk 3]
```diff
@@ -44,11 +38,61 @@ Interoperability with packages from the wider scientific Python ecosystem comes
...
+        :classes: btn-outline-primary btn-block
+    ---
+    Learn how to use Iris.
+    +++
+    .. link-button:: user_guide_index
+        :type: ref
+        :text: User Guide
+        :classes: btn-outline-primary btn-block
+    ---
+    Documentation for the Iris Application Programming Interface.
```
Comment (rcomer) [inline docs/iris/src/index.rst:77]: I'm not sure whether API is a familiar term for a lot of our target user-base.  Maybe something like "Browse full Iris functionality by module"?

File: docs/iris/src/index.rst [hunk 4]
```diff
@@ -44,11 +38,61 @@ Interoperability with packages from the wider scientific Python ecosystem comes
...
+    :column: col-lg-4 col-md-4 col-sm-6 col-xs-12 p-2
+
+    Install Iris to use or for development.
+    +++
+    .. link-button:: installing_iris
+        :type: ref
+        :text: Installing Iris
+        :classes: btn-outline-primary btn-block
+    ---
+    View the gallery that includes python code used to create it.
```
Comment (rcomer) [inline docs/iris/src/index.rst:56]: How about something like "Example code to create a variety of plots"?

**Review by rcomer — COMMENTED** [review]
Definitely like the buttons.  I just have some suggestions for some of the descriptions which you are free to adapt  or ignore as you see fit!

**Comment by trexfeathers** [comment]
Love it!

File: docs/iris/src/index.rst (hunk 1 above)
Comment (trexfeathers) [reply docs/iris/src/index.rst:63]: To be fair, all of it has been recently added to Iris, but only some of it has been released so far.

Comment (rcomer) [reply docs/iris/src/index.rst:63]: In fact, some things have be _removed_ from Iris....

Comment (tkknight) [reply docs/iris/src/index.rst:63]: how about:

> "Find out what has recently changed in Iris."

Comment (trexfeathers) [reply docs/iris/src/index.rst]: >"Find out what has recently changed in Iris."

I'm in favour of this

**Comment by tkknight** [comment]
Review comments addressed via https://github.com/SciTools/iris/pull/3884/commits/0d47a55fe3dc26f4def67655d5c0476c61dd0e14

**Comment by tkknight** [comment]
anymore comments?  if not could someone please review and approve please.

**Review by pp-mo — APPROVED** [review]
A big step forward

**Comment by bjlittle** [comment]
@tkknight Could you just make sure that you've serviced all the reviewer comments, then I'll bank this 👍

**Comment by rcomer** [comment]
Think mine were the only comments, so now banked.

**Pull request merged**